src/lib/y2storage/autoinst_issues/invalid_value.rb

```diff
@@ -26,28 +26,32 @@ module AutoinstIssues
     # Represents an AutoYaST situation where an invalid value was given.
     #
     # @example Invalid value 'auto' for attribute :size on /home partition
-    #   problem = InvalidValue.new("/home", :size, "auto")
+    #   section = AutoinstProfile::PartitioningSection.new_from_hashes({"size" => "auto"})
+    #   problem = InvalidValue.new(section, :size)
+    #   problem.value #=> "auto"
+    #   problem.attr  #=> :size
     class InvalidValue < Issue
-      # @return [String] Device affected by this error
-      attr_reader :device
       # @return [Symbol] Name of the missing attribute
       attr_reader :attr
-      # @return [Object] Invalid value
-      attr_reader :value
       # @return [Object] New value or :skip to skip the section.
       attr_reader :new_value
 
-      # @param device    [String] Device (`/`, `/dev/sda`, etc.)
-      # @param attr      [Symbol] Name of the missing attribute
-      # @param value     [Integer,String,Symbol] Invalid value
+      # @param section   [Object] Section where it was detected (see {AutoinstProfile})
+      # @param attr      [Symbol] Name of the invalid attribute
       # @param new_value [Integer,String,Symbol] New value or :skip to skip the section
-      def initialize(device, attr, value, new_value = :skip)
-        @device = device
+      def initialize(section, attr, new_value = :skip)
+        @section = section
         @attr = attr
-        @value = value
         @new_value = new_value
       end
 
+      # Return the invalid value
+      #
+      # @return [Integer,String,Symbol] Invalid value
+      def value
+        section.send(attr)
```

Comment:
well, this allow to also call private methods, which is not so nice from my POV. `public_send` is not enough?